### ISSUE TITLE
support for nonstandard yaml-cpp installation locations

### ISSUE THREAD

**Issue by davidlange6** [bkryza/clang-uml#206]
I am trying out clang-uml on a linux system with yaml-cpp installed in a nonstandard place. I needed the following change to get the build to properly include the corresponding -I for the build to find my yaml-cpp headers

```
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9b170746..d1d9d3b9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -138,7 +138,8 @@ set(THIRDPARTY_HEADERS_DIR ${PROJECT_SOURCE_DIR}/thirdparty/)
 #
 include_directories(${LLVM_INCLUDE_DIRS})
 include_directories(${CLANG_UML_INSTALL_INCLUDE_DIR})
-include_directories(${YAML_CPP_INCLUDE_DIRS})
+#was with an s...
+include_directories(${YAML_CPP_INCLUDE_DIR})
 include_directories(${UML_HEADERS_DIR})
 include_directories(${THIRDPARTY_HEADERS_DIR})
 include_directories(${PROJECT_SOURCE_DIR}/src/)
```

Since I see the opposite change was made last year for Mac support (https://github.com/bkryza/clang-uml/commit/b06634842915a1ec89c324648c61e5d43ea8c306), I didn't open a PR to fix this..

**Comment by bkryza** [comment]
@davidlange6 Thanks for reporting this. I'll try to reproduce it and fix as needed...

**Comment by agent00jackson** [comment]
Might be relevant, I just tried to install via conda and am getting this error even though yaml-cpp is installed:
```
clang-uml: error while loading shared libraries: libyaml-cpp.so.0.7: cannot open shared object file: No such file or directory
```

Environment:
```
# packages in environment at /home/n0210f/mambaforge/envs/uml:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                       2_gnu    conda-forge
bzip2                     1.0.8                hd590300_5    conda-forge
ca-certificates           2023.7.22            hbcca054_0    conda-forge
clang-uml                 0.3.1                h3fd9d12_0    bkryza/label/clang-uml
icu                       73.2                 h59595ed_0    conda-forge
ld_impl_linux-64          2.40                 h41732ed_0    conda-forge
libclang                  15.0.6          default_h2e3cab8_0    conda-forge
libclang-cpp              15.0.6          default_h2e3cab8_0    conda-forge
libclang-cpp15            15.0.6          default_h2e3cab8_0    conda-forge
libclang13                15.0.6          default_h3a83d3e_0    conda-forge
libffi                    3.4.2                h7f98852_5    conda-forge
libgcc-ng                 13.2.0               h807b86a_2    conda-forge
libgomp                   13.2.0               h807b86a_2    conda-forge
libiconv                  1.17                 h166bdaf_0    conda-forge
libllvm15                 15.0.7               h5cf9203_3    conda-forge
libnsl                    2.0.1                hd590300_0    conda-forge
libsqlite                 3.44.0               h2797004_0    conda-forge
libstdcxx-ng              13.2.0               h7e041cc_2    conda-forge
libuuid                   2.38.1               h0b41bf4_0    conda-forge
libxml2                   2.11.5               h232c23b_1    conda-forge
libzlib                   1.2.13               hd590300_5    conda-forge
ncurses                   6.4                  h59595ed_2    conda-forge
openssl                   3.1.4                hd590300_0    conda-forge
pip                       23.3.1             pyhd8ed1ab_0    conda-forge
python                    3.10.13         hd12c33a_0_cpython    conda-forge
readline                  8.2                  h8228510_1    conda-forge
setuptools                68.2.2             pyhd8ed1ab_0    conda-forge
tk                        8.6.13          noxft_h4845f30_101    conda-forge
tzdata                    2023c                h71feb2d_0    conda-forge
wheel                     0.41.3             pyhd8ed1ab_0    conda-forge
xz                        5.2.6                h166bdaf_0    conda-forge
yaml-cpp                  0.7.0                h59595ed_3    conda-forge
zstd                      1.5.5                hfc55251_0    conda-forge
```

**Comment by bkryza** [comment]
@davidlange6 I've included your change for yaml-cpp include dir path, and I've also added a CMake option `CMAKE_PREFIX` where you can provide a path to where the yaml-cpp is installed,  (it has to contain `yaml-cpp-config.cmake` which is installed by yaml-cpp cmake automatically...)

As to the conda issue, I've tried to reproduce it by I'm currently getting another error:

```bash
❯ ~/miniconda3/bin/clang-uml --version
/home/bartek/miniconda3/bin/clang-uml: /lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.32' not found (required by /home/bartek/miniconda3/bin/clang-uml)
```

I'll try to fix it in next release...

**Comment by davidlange6** [comment]
Yes, this fixes my setup. Thanks!

**Comment by CsatiZoltan** [comment]
@bkryza I got the same error as @agent00jackson. Steps to reproduce:

```bash
$ mamba create --name clang-uml
Empty environment created at prefix: <user_name>/Programs/miniforge3/envs/clang-uml
$ mamba activate clang-uml
$ mamba install -c bkryza/label/clang-uml clang-uml
bkryza/label/clang-uml/linux-64                     ??.?MB @  ??.?MB/s  0.2s
bkryza/label/clang-uml/noarch                      116.0 B @ 111.0 B/s  1.3s
conda-forge/noarch                                  17.9MB @   4.7MB/s 13.2s
conda-forge/linux-64                                40.8MB @   2.6MB/s 42.1s


Transaction

  Prefix: <user_name>/Programs/miniforge3/envs/clang-uml

  Updating specs:

   - clang-uml


  Package             Version  Build               Channel                      Size
──────────────────────────────────────────────────────────────────────────────────────
  Install:
──────────────────────────────────────────────────────────────────────────────────────

  + _libgcc_mutex         0.1  conda_forge         conda-forge                Cached
  + _openmp_mutex         4.5  2_gnu               conda-forge                Cached
  + clang-uml           0.5.6  h3fd9d12_0          bkryza/label/clang-uml        2MB
  + libclang           19.1.4  default_hb5137d0_0  conda-forge                  24kB
  + libclang-cpp       19.1.4  default_hb5137d0_0  conda-forge                  24kB
  + libclang-cpp19.1   19.1.4  default_hb5137d0_0  conda-forge                  21MB
  + libclang13         19.1.4  default_h9c6a7e4_0  conda-forge                  12MB
  + libgcc             14.2.0  h77fa898_1          conda-forge                Cached
  + libgcc-ng          14.2.0  h69a702a_1          conda-forge                Cached
  + libgomp            14.2.0  h77fa898_1          conda-forge                Cached
  + libiconv             1.17  hd590300_2          conda-forge                Cached
  + libllvm19          19.1.6  ha7bfdaf_0          conda-forge                  40MB
  + liblzma             5.6.3  hb9d3cd8_1          conda-forge                 111kB
  + libstdcxx          14.2.0  hc0a3c3a_1          conda-forge                Cached
  + libstdcxx-ng       14.2.0  h4852527_1          conda-forge                  54kB
  + libxml2            2.13.5  h0d44e9d_1          conda-forge                 690kB
  + libzlib             1.3.1  hb9d3cd8_2          conda-forge                Cached
  + yaml-cpp            0.8.0  h59595ed_0          conda-forge                Cached
  + zstd                1.5.6  ha6fb4c9_0          conda-forge                Cached

  Summary:

  Install: 19 packages

  Total download: 76MB

$ clang-uml --init
clang-uml: error while loading shared libraries: libyaml-cpp.so.0.8: cannot open shared object file: No such file or directory
```

**Comment by bkryza** [comment]
@CsatiZoltan Thanks for reporting this, I'll try to fix it when releasing a new version next month. It did work 3 weeks ago when I was releasing the latest version as I always check if it does...

**Comment by bkryza** [comment]
@CsatiZoltan Actually, I've just rebuild and published the latest release `0.5.6` on conda and it seems to have fixed it:

```console
❯ docker run -it continuumio/miniconda3 bash
(base) root@9a131c2ce78c:/# conda create --name clang-uml
...
(base) root@9a131c2ce78c:/# conda activate clang-uml
(clang-uml) root@9a131c2ce78c:/# conda install -c conda-forge -c bkryza/label/clang-uml clang-uml
Channels:
 - conda-forge
 - bkryza/label/clang-uml
 - defaults
Platform: linux-64
Collecting package metadata (repodata.json): done
Solving environment: done

## Package Plan ##

  environment location: /opt/conda/envs/clang-uml

  added / updated specs:
    - clang-uml


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    _libgcc_mutex-0.1          |      conda_forge           3 KB  conda-forge
    _openmp_mutex-4.5          |            2_gnu          23 KB  conda-forge
    clang-uml-0.5.6            |       h3fd9d12_0         2.3 MB  bkryza/label/clang-uml
    libclang-19.1.4            |default_hb5137d0_0          23 KB  conda-forge
    libclang-cpp-19.1.4        |default_hb5137d0_0          23 KB  conda-forge
    libclang-cpp19.1-19.1.4    |default_hb5137d0_0        19.6 MB  conda-forge
    libclang13-19.1.4          |default_h9c6a7e4_0        11.3 MB  conda-forge
    libgcc-14.2.0              |       h77fa898_1         829 KB  conda-forge
    libgcc-ng-14.2.0           |       h69a702a_1          53 KB  conda-forge
    libgomp-14.2.0             |       h77fa898_1         450 KB  conda-forge
    libiconv-1.17              |       hd590300_2         689 KB  conda-forge
    libllvm19-19.1.6           |       ha7bfdaf_0        38.3 MB  conda-forge
    liblzma-5.6.3              |       hb9d3cd8_1         109 KB  conda-forge
    libstdcxx-14.2.0           |       hc0a3c3a_1         3.7 MB  conda-forge
    libstdcxx-ng-14.2.0        |       h4852527_1          53 KB  conda-forge
    libxml2-2.13.5             |       h0d44e9d_1         674 KB  conda-forge
    libzlib-1.3.1              |       hb9d3cd8_2          60 KB  conda-forge
    yaml-cpp-0.8.0             |       h59595ed_0         200 KB  conda-forge
    zstd-1.5.6                 |       ha6fb4c9_0         542 KB  conda-forge
    ------------------------------------------------------------
                                           Total:        78.8 MB

The following NEW packages will be INSTALLED:

  _libgcc_mutex      conda-forge/linux-64::_libgcc_mutex-0.1-conda_forge 
  _openmp_mutex      conda-forge/linux-64::_openmp_mutex-4.5-2_gnu 
  clang-uml          bkryza/label/clang-uml/linux-64::clang-uml-0.5.6-h3fd9d12_0 
  libclang           conda-forge/linux-64::libclang-19.1.4-default_hb5137d0_0 
  libclang-cpp       conda-forge/linux-64::libclang-cpp-19.1.4-default_hb5137d0_0 
  libclang-cpp19.1   conda-forge/linux-64::libclang-cpp19.1-19.1.4-default_hb5137d0_0 
  libclang13         conda-forge/linux-64::libclang13-19.1.4-default_h9c6a7e4_0 
  libgcc             conda-forge/linux-64::libgcc-14.2.0-h77fa898_1 
  libgcc-ng          conda-forge/linux-64::libgcc-ng-14.2.0-h69a702a_1 
  libgomp            conda-forge/linux-64::libgomp-14.2.0-h77fa898_1 
  libiconv           conda-forge/linux-64::libiconv-1.17-hd590300_2 
  libllvm19          conda-forge/linux-64::libllvm19-19.1.6-ha7bfdaf_0 
  liblzma            conda-forge/linux-64::liblzma-5.6.3-hb9d3cd8_1 
  libstdcxx          conda-forge/linux-64::libstdcxx-14.2.0-hc0a3c3a_1 
  libstdcxx-ng       conda-forge/linux-64::libstdcxx-ng-14.2.0-h4852527_1 
  libxml2            conda-forge/linux-64::libxml2-2.13.5-h0d44e9d_1 
  libzlib            conda-forge/linux-64::libzlib-1.3.1-hb9d3cd8_2 
  yaml-cpp           conda-forge/linux-64::yaml-cpp-0.8.0-h59595ed_0 
  zstd               conda-forge/linux-64::zstd-1.5.6-ha6fb4c9_0 


Proceed ([y]/n)? y


Downloading and Extracting Packages:
                                                                                                                                                                                                                                                                      
Preparing transaction: done                                                                                                                                                                                                                                           
Verifying transaction: done                                                                                                                                                                                                                                           
Executing transaction: done                                                                                                                                                                                                                                           

(clang-uml) root@9a131c2ce78c:/# clang-uml --init                                                                                                                                                                                                                     

(clang-uml) root@9a131c2ce78c:/# cat .clang-uml
# Change to directory where compile_commands.json is                                                                                                                                                                                                                  
compilation_database_dir: .                                                                                                                                                                                                                                           
# Change to directory where diagram should be written                                                                                                                                                                                                                 
output_directory: docs/diagrams                                                                                                                                                                                                                                       
diagrams:                                                                                                                                                                                                                                                             
  example_class_diagram:                                                                                                                                                                                                                                              
    type: class                                                                                                                                                                                                                                                       
    glob:                                                                                                                                                                                                                                                             
      - src/*.cpp                                                                                                                                                                                                                                                     
    using_namespace:                                                                                                                                                                                                                                                  
      - myproject                                                                                                                                                                                                                                                     
    include:                                                                                                                                                                                                                                                          
      namespaces:                                                                                                                                                                                                                                                     
        - myproject
    exclude:
      namespaces:
        - myproject::detail
```

Can you try again?

**Comment by CsatiZoltan** [comment]
Thanks, it works now. Maybe you forgot updating the version label in the conda repo (it still shows 0.3.9)?
